moncli/types.py

```diff
@@ -211,4 +214,78 @@ def validate_timezone(self, value):
         try:
             pytz.timezone(value)
         except (UnknownTimeZoneError):
-            raise ValidationError('Unknown time zone "{}".'.format(value))
\ No newline at end of file
+            raise ValidationError('Unknown time zone "{}".'.format(value))
+
+class StatusType(MondayType):
+    native_type = str
+    allow_casts = (int, str)
+    null_value = {}
+    def __init__(self, id: str = None, title: str = None,data_mapping:dict =None ,  *args, **kwargs):
+        if data_mapping:
+            values= [value for value in data_mapping.values()]        
+            self.native_type = values[0].__class__
+            self.choices = values
+        self._data_mapping =  data_mapping 
+        super().__init__(id=id, title=title, *args, **kwargs)
+    
+    def _process(self, value):
+        labels = self.metadata['labels']
+        if self.native_type == str:
+            try:
+                if isinstance(int(value),int):
```

Comment:
I would remove the isinstance check here and simply try to cast the value as an int without assigning it to anything.  If the value can be casted as an int, then we know we are dealing with an index.  If the cast throws a value error, then we are likely dealing with a label directly and the value should be passed below to the next check.